cpp/modmesh/python/wrapper/modmesh/TypeBroadcast.hpp

```diff
@@ -104,7 +104,7 @@ struct TypeBroadcastImpl
             sidx_init[i] = 0;
         }
 
-        copy_idx(arr_out, slices, arr_new, left_shape, sidx_init, arr_out.ndim() - 1);
+        copy_idx(arr_out, slices, arr_new, left_shape, sidx_init, static_cast<int>(arr_out.ndim()) - 1);
```

Comment:
Why not change the signature of the lambda `copy_idx`?

Reply:
we need to make it `int` since there is a ` if ( dim < 0 ) `

Reply:
Oh, I missed that.